Mathlib/Geometry/Manifold/ChartedSpace.lean

```diff
@@ -622,6 +622,19 @@ theorem chart_target_mem_nhds (x : M) : (chartAt H x).target ∈ 𝓝 (chartAt H
   (chartAt H x).open_target.mem_nhds <| mem_chart_target H x
 #align chart_target_mem_nhds chart_target_mem_nhds
 
+variable (M) in
+@[simp]
+theorem iUnion_source_eq_univ : (⋃ x : M, (chartAt H x).source) = (univ : Set M) :=
+  eq_univ_iff_forall.mpr fun x ↦ mem_iUnion.mpr ⟨x, mem_chart_source H x⟩
+
+theorem isOpen_iff (s : Set M) :
+    IsOpen s ↔ ∀ x : M, IsOpen <| chartAt H x '' ((chartAt H x).source ∩ s) := by
```

Comment:
What do you think about adding the following lemma somewhere in `Topology.Basic`?

```lean
lemma isOpen_iff_of_cover {ι} {f : ι → Set X} (ho : ∀ i, IsOpen (f i)) (hU : (⋃ i, f i) = univ) {s : Set X} :
    IsOpen s ↔ ∀ i, IsOpen (f i ∩ s)
```

Reply:
That sounds like a good idea; will do so (not today any more).

Reply:
Used that lemma in ChartedSpace: is clearer, but not much shorter.